docs/task_settings.rst

```diff
@@ -1,10 +1,127 @@
-To use the S3 repository, please set ``s3://{YOUR_REPOSITORY_NAME}`` to :attr:`~gokart.task.TaskOnKart.workspace_directory`,
-and appropriate values to the parameters of :class:`~gokart.s3_config.S3Config`.
\ No newline at end of file
+If you would like to use local directory, please set a local directory path to :attr:`~gokart.task.TaskOnKart.workspace_directory`. Please refer to :doc:`task_parameters` for how to set it up.
+
+It is recommended to use the config file since it does not change much.
+
+::
+
+    # base.ini
+    [TaskOnKart]
+    workspace_directory=%(TASK_WORKSPACE_DIRECTORY)s
+
+.. code:: python
+
+    # main.py
+    import gokart
+    gokart.add_config('base.ini')
+
+
+To use the S3 or GCS repository, please set the bucket path as ``s3://{YOUR_REPOSITORY_NAME}`` or ``gs://{YOUR_REPOSITORY_NAME}`` respectively.
+
+If use S3 or GCS, please set credential information to Environment Variables.
+
+.. code:: sh
+
+    # S3
+    export AWS_ACCESS_KEY_ID='~~~'  # AWS access key
+    export AWS_SECRET_ACCESS_KEY='~~~'  # AWS secret access key
+
+    # GCS
+    export GCS_CREDENTIAL='~~~'  # GCS credential
+    export DISCOVER_CACHE_LOCAL_PATH='~~~'  # The local file path of discover api cache.
+
+
+Rerun task
+----------
+
+There are times when we want to rerun a task, such as when change script or on batch. Please use the ``rerun`` parameter or add an arbitrary parameter.
+
+
+When set rerun as follows:
+
+.. code:: python
+
+    # rerun TaskA
+    gokart.build(Task(rerun=True))
+
+
+When used from an argument as follows:
+
+.. code:: python
+
+    # main.py
+    class Task(gokart.TaskOnKart):
+        def run(self):
+            self.dump('hello')
+
+.. code:: sh
+
+    python main.py Task --local-scheduler --rerun
+
+
+``rerun`` parameter will look at the dependent tasks up to one level.
+
+Example: Suppose we have a straight line pipeline composed of TaskA, TaskB and TaskC,  and TaskC is an endpoint of this pipeline. We also suppose that all the tasks have already been executed.
+
+- TaskA(rerun=True)  ->  TaskB  ->  TaskC    # not rerunning
+- TaskA  ->  TaskB(rerun=True)  ->  TaskC    # reruning TaskB and TaskC
+
+This is due to the way intermediate files are handled. ``rerun`` parameter is ``significant=False``, it does not affect the hash value. It is very important to understand this difference.
+
+
+If you want to change the parameter of TaskA and rerun TaskB and TaskC, recommend adding an arbitrary parameter.
+
+.. code:: python
+
+    class TaskA(gokart.TaskOnKart):
+        __version = luigi.IntParameter(default=1)
+
+If the hash value of TaskA will change, the dependent tasks (in this case, TaskB and TaskC) will rerun.
+
+
+Fix random seed
+---------------
+
+Every Task has a parameter named :attr:`~gokart.task.TaskOnKart.fix_random_seed_methods` and :attr:`~gokart.task.TaskOnKart.fix_random_seed_value`. This can be used to fix the random seed.
```

Comment:
[nits]
```suggestion
Every task has a parameter named :attr:`~gokart.task.TaskOnKart.fix_random_seed_methods` and :attr:`~gokart.task.TaskOnKart.fix_random_seed_value`. This can be used to fix the random seed.
```